yt_dlp/utils/traversal.py

```diff
@@ -168,14 +169,38 @@ def apply_key(key, obj, is_last):
                 result = next((v for k, v in obj.groupdict().items() if casefold(k) == key), None)
 
         elif isinstance(key, (int, slice)):
-            if is_iterable_like(obj, collections.abc.Sequence):
+            if is_iterable_like(obj, collections.abc.Sequence) or isinstance(obj, xml.etree.ElementTree.Element):
                 branching = isinstance(key, slice)
                 with contextlib.suppress(IndexError):
                     result = obj[key]
             elif traverse_string:
                 with contextlib.suppress(IndexError):
                     result = str(obj)[key]
 
+        elif isinstance(obj, xml.etree.ElementTree.Element) and isinstance(key, str):
+            xpath, sep, special = key.rpartition('/')
+            has_specials = special.startswith('@') or special == 'text()'
+            if not has_specials:
+                xpath = f'{xpath}{sep}{special}'
+
+            # Allow abbreviations of relative paths, absolute paths error
+            if xpath.startswith('/'):
+                xpath = f'.{xpath}'
+            elif xpath and not xpath.startswith('./'):
+                xpath = f'./{xpath}'
+
+            findings = obj.iterfind(xpath) if xpath else [obj]
+            if has_specials:
+                result = [
+                    element.attrib if special == '@'
+                    else try_call(element.attrib.get, args=(special[1:],)) if special.startswith('@')
+                    else element.text if special == 'text()'
+                    else None
+                    for element in findings
+                ]
+            else:
+                result = list(findings)
+
```

Comment:
I would be for only returning the value instead of a list, when `@attr` ot `text()` are used without a leading xpath.
Pretty much in every case that you pass a sole `@attr` you would want to retrieve the actual value and not a list with the value. Currently, you need to do this:
```python
traverse_obj(etree, ('@name', 0)) # string 
traverse_obj(etree, ('text()', 0)) # string
```
Adding something like this should work to only unpack the list when no xpath was evaluated, which in turn is only the case if just a special key is present.
```suggestion
if not xpath:
    result = result[0]

```